arangod/Utils/Cursor.h

```diff
@@ -85,6 +92,30 @@ class Cursor {
 
   void setDeleted() { _isDeleted = true; }
 
+  void setLastQueryBatchObject(
+      std::shared_ptr<velocypack::Buffer<uint8_t>> buffer) {
+    _currentBatchResult.second = buffer;
+  }
+
+  Result getLastBatchResult(std::string const& batchId,
+                            VPackBufferUInt8& buffer) {
+    if (_currentBatchResult.first.compare(batchId) != 0) {
+      return Result(TRI_ERROR_HTTP_NOT_FOUND, "batch id not found");
+    }
+    VPackBuilder builder(*(_currentBatchResult.second.get()));
```

Comment:
`_currentBatchResult` contains a shared_ptr to the Buffer already. Why not simply return it? Creating another Builder on top is just unnecessary overhead.